### DISCUSSION TITLE
Any ideas how to catch a UI bug of the settings-window, it freezes sometimes

### DISCUSSION THREAD

**Discussion by pc223** [Flow-Launcher/Flow.Launcher#543] [Q&A] [answered]
I notice the settings-window sometimes freeze when I try to open. Can see the settings content but can't interact, still able to close the settings-window. When close and reopen, it works again. No exception in the logs, it tends to happen after a long run time, maybe a couple of days, so really hard to reproduce.

I'm trying to investigate but not sure where to put the logging. Any ideas how to debug this?

Just add a bunch of logs in this `SingletonWindowOpener`, see if it'll stop in any step

<img src="https://user-images.githubusercontent.com/10551242/124297181-ff47e580-db84-11eb-8edb-19309ff991a2.png" width="400">

**Comment by taooceros** [comment, marked as answer]
Check out this https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/issues/559

I think that's reproducible, though I don't understand the main reason.

**Reply by pc223** [reply]
Oh so lock screen is the way to reproduce the freezes haha 😄 , so you guys aware of this bug for long time?

**Reply by taooceros** [reply]
No, I just discover and confirm it on google when this discussion is posted.

**Reply by taooceros** [reply]
By searching wpf window freeze🤣

**Reply by pc223** [reply]
> No, I just discover and confirm it on google when this discussion is posted.

Nice! Seems we have a good chance to fix this, I feared we have to take much longer time to pin point this 😆

**Reply by taooceros** [reply]
I found a discussion with solution, and it is easy. 

Here's the link https://forums.ni.com/t5/Measurement-Studio-for-NET/Window-with-WPF-Graph-freezes-when-windows-account-locked/td-p/2419516/page/2?profile.language=zh-CN

![image](https://user-images.githubusercontent.com/45326534/124305584-e09f1a00-db97-11eb-8c4a-22c7818f2335.png)

**Reply by taooceros** [reply]
https://stackoverflow.com/questions/4951058/software-rendering-mode-wpf

**Reply by pc223** [reply]
> I found a discussion with solution, and it is easy.
> 
> Here's the link https://forums.ni.com/t5/Measurement-Studio-for-NET/Window-with-WPF-Graph-freezes-when-windows-account-locked/td-p/2419516/page/2?profile.language=zh-CN
> 
> ![image](https://user-images.githubusercontent.com/45326534/124305584-e09f1a00-db97-11eb-8c4a-22c7818f2335.png)

👍  I'll try them out, maybe make a PR if it works 😼

**Comment by taooceros** [comment]
It would be better to check or add Log in SettingsWindow.xaml.cs if you would like to debug this, since that will allow more access to private property.

**Comment by jjw24** [comment]
good catch, this actually happens with prd release

**Comment by jjw24** [comment]
work around fix:
https://github.com/Flow-Launcher/Flow.Launcher/pull/544
